modules/imgproc/src/drawing.cpp

```diff
@@ -1299,15 +1299,11 @@ CollectPolyEdges( Mat& img, const Point2l* v, int count, std::vector<PolyEdge>&
                 if (t0.y != t1.y)
                 {
                     pt0c.y = t0.y; pt1c.y = t1.y;
-                    pt0c.x = (int64)(t0.x) << XY_SHIFT;
```

Comment:
Does any test check clipping scenario? Non-zero shift?

Reply:
I did not add tests for all scenarios since `fillPoly` and `polylines` handle incoming points slightly different and also have differences in drawing the contours for certain parameters. To make the contours match in all scenarios probably both functions would need further adjustments.

This PR should fix the matching contours for `line_type < LINE_AA` and zero shift since it used to match before the changes of #23076